### PR TITLE
Fix testbed lock race

### PR DESCRIPTION
**Pull request by jablonskim**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -83,50 +82,49 @@ pipeline {
             steps {
                 script {
                     try {
-                        lock(dataNetwork) {
-                            // 'Provision' stage
+                        lock('vmware-lock') {
```
Comment (kaweue) [inline Jenkinsfile]: Please check if jobs take slots.

Comment (kaweue) [reply Jenkinsfile]: Try to split this 'Provision & Deploy & Test' into 3 stages

**Review by kaweue — DISMISSED** [review]

File: ansible/wait_for_folder.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    subfolder = folder.CreateFolder(args.name)
+    if not subfolder:
+        raise Exception("Couldn't create the Folder with the provided name "
+                        "'{}'".format(args.name))
+
+    return 0
+
+
+if __name__ == "__main__":
+    exit(main())
```
Comment (krdln) [inline ansible/wait_for_folder.py]: Perhaps we can merge this two files and use something like command argument?
Eg.
```
python vm_folder.py --datacenter foo ... wait
```
and 
```
python vm_folder.py --datacenter foo ... remove --folder bar
```

**Comment by jablonskim** [comment]
recheck no bug

**Comment by jablonskim** [comment]
recheck no bug

**Comment by jablonskim** [comment]
recheck no bug

**Comment by jablonskim** [comment]
recheck no bug

File: Jenkinsfile [hunk 3]
```diff
@@ -65,100 +81,146 @@ pipeline {
...
 
-        // Variables are not supported in declarative pipeline.
-        // Possible workaround: store SpawnedTestbedVMNames in stashed file.
-        // def SpawnedTestbedVMNames = ''
-
-        // NOTE: Currently nesting multiple stages in lock directive is unsupported
-        stage('Provision & Deploy & Test') {
-            agent none
+        stage('Lock') {
+            agent { label 'ansible' }
```
Comment (sodar) [inline Jenkinsfile:85]: Should we use `renter` instead of `ansible` here?

**Review by sodar — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 3 above)
Comment (jablonskim) [reply Jenkinsfile:85]: Yes, but after PR with creating and deploying `renter` VM. For now we have to use `ansible`

**Review by sodar — APPROVED** [review]

**Pull request merged**